homeassistant/components/here_travel_time/sensor.py

```diff
@@ -0,0 +1,334 @@
+        vol.Optional(
+            CONF_ROUTE_MODE, default=ROUTE_MODE_FASTEST
+        ): vol.In(ROUTE_MODE),
+        vol.Optional(CONF_TRAFFIC_MODE, default=False): cv.boolean,
+        vol.Optional(CONF_UNIT_SYSTEM): vol.In(UNITS),
+    }
+)
+
+TRACKABLE_DOMAINS = ['device_tracker', 'sensor', 'zone', 'person']
+DATA_KEY = 'here_travel_time'
+
+NO_ROUTE_ERROR_MESSAGE = "HERE could not find a route based on the input"
+
+
+def convert_time_to_utc(timestr):
+    """Take a string like 08:00:00 and convert it to a unix timestamp."""
+    combined = datetime.combine(
+        dt_util.start_of_local_day(), dt_util.parse_time(timestr)
+    )
+    if combined < datetime.now():
+        combined = combined + timedelta(days=1)
+    return dt_util.as_timestamp(combined)
+
+
+def setup_platform(hass, config, add_entities_callback, discovery_info=None):
+    """Set up the HERE travel time platform."""
+    def run_setup(event):
+        """
+        Delay the setup until Home Assistant is fully initialized.
+
+        This allows any entities to be created already
+        """
+        hass.data.setdefault(DATA_KEY, [])
+
+        app_id = config[CONF_APP_ID]
+        app_code = config[CONF_APP_CODE]
+        origin = config[CONF_ORIGIN]
+        destination = config[CONF_DESTINATION]
+
+        travel_mode = config.get(CONF_MODE)
+        traffic_mode = config.get(CONF_TRAFFIC_MODE)
+        route_mode = config.get(CONF_ROUTE_MODE)
+        name = config.get(CONF_NAME, DEFAULT_NAME)
+        units = config.get(CONF_UNIT_SYSTEM, hass.config.units.name)
+
+        here_data = HERETravelTimeData(None,
+                                       None,
+                                       app_id,
+                                       app_code,
+                                       travel_mode,
+                                       traffic_mode,
+                                       route_mode,
+                                       units)
+
+        sensor = HERETravelTimeSensor(hass,
+                                      name,
+                                      origin,
+                                      destination,
+                                      here_data)
+
+        hass.data[DATA_KEY].append(sensor)
+
+        add_entities_callback([sensor])
+
+    # Wait until start event is sent to load this component.
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, run_setup)
+
+
+class HERETravelTimeSensor(Entity):
+    """Representation of a HERE travel time sensor."""
+
+    def __init__(self, hass, name, origin, destination, here_data):
+        """Initialize the sensor."""
+        self._hass = hass
+        self._name = name
+        self._here_data = here_data
+        self._unit_of_measurement = 'min'
+
+        # Check if location is a trackable entity
+        if origin.split('.', 1)[0] in TRACKABLE_DOMAINS:
+            self._origin_entity_id = origin
+        else:
+            self._here_data.origin = origin
```

Comment:
Make sure to initialize all attributes regardless of case. Set them to `None` at minimum.